REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/security/KeyUtils.java

```diff
@@ -0,0 +1,127 @@
+package com.epam.esm.security;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ObjectUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.security.*;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.EncodedKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Arrays;
+
+@Component
+@Slf4j
+public class KeyUtils {
+
+    private final Environment environment;
+
+    @Value("${access-token.private}")
+    private String accessTokenPrivateKeyPath;
+
+    @Value("${access-token.public}")
+    private String accessTokenPublicKeyPath;
+
+    @Value("${refresh-token.private}")
+    private String refreshTokenPrivateKeyPath;
+
+    @Value("${refresh-token.public}")
+    private String refreshTokenPublicKeyPath;
+
+    private KeyPair accessTokenKeyPair;
+    private KeyPair refreshTokenKeyPair;
+
+    @Autowired
+    public KeyUtils(Environment environment) {
+        this.environment = environment;
+    }
+
+    private KeyPair getAccessTokenKeyPair() {
+        if (ObjectUtils.isEmpty(accessTokenKeyPair)) {
+            accessTokenKeyPair = getTokenKeyPair(accessTokenPrivateKeyPath, accessTokenPublicKeyPath);
+        }
+        return accessTokenKeyPair;
+    }
+
+    private KeyPair getRefreshTokenKeyPair() {
+        if (ObjectUtils.isEmpty(refreshTokenKeyPair)) {
+            refreshTokenKeyPair = getTokenKeyPair(refreshTokenPrivateKeyPath, refreshTokenPublicKeyPath);
+        }
+        return refreshTokenKeyPair;
+    }
+
+    private KeyPair getTokenKeyPair(String privateKeyPath, String publicKeyPath) {
+        KeyPair keyPair;
+        File publicKeyFile = new File(publicKeyPath);
```

Comment:
This method is too big, split it please.

Reply:
Done